--- _CoqProject.orig	2025-12-26 00:00:00.000000000 +0100
+++ _CoqProject	2025-12-26 00:00:00.000000000 +0100
@@ -1 +1,21 @@
 -Q . LF
+Preface.v
+Basics.v
+Induction.v
+Lists.v
+Poly.v
+Tactics.v
+Logic.v
+IndProp.v
+Maps.v
+ProofObjects.v
+IndPrinciples.v
+Rel.v
+Imp.v
+ImpParser.v
+ImpCEvalFun.v
+Extraction.v
+Auto.v
+AltAuto.v
+Postscript.v
+Bib.v
--- Makefile.coq.orig	2025-12-26 00:00:00.000000000 +0100
+++ Makefile.coq	2025-12-26 00:00:00.000000000 +0100
@@ -541,7 +541,7 @@
 	$(SHOW)'COQDOC -pdf $(GAL)'
 	$(HIDE)$(COQDOC) \
 		-toc $(COQDOCFLAGS) -pdf $(GAL) $(COQDOCLIBS) \
-		-o $@ `$(COQDEP) -sort $(VFILES)`
+		-o $@ Preface.v Basics.v Induction.v Lists.v Poly.v Tactics.v Logic.v IndProp.v Maps.v ProofObjects.v IndPrinciples.v Rel.v Imp.v ImpParser.v ImpCEvalFun.v Extraction.v Auto.v AltAuto.v Postscript.v Bib.v

 # FIXME: not quite right, since the output name is different
 gallinahtml: GAL=-g
--- Bib.v.orig	2025-12-26 15:07:45.023944728 +0100
+++ Bib.v	2025-12-26 15:07:47.051959963 +0100
@@ -8,27 +8,27 @@
 [Bertot 2004] Interactive Theorem Proving and Program Development:
   Coq'Art: The Calculus of Inductive Constructions, by Yves Bertot and
   Pierre Casteran.  Springer-Verlag, 2004.
-  {https://tinyurl.com/z3o7nqu}
+  https://tinyurl.com/z3o7nqu

 [Chlipala 2013] Certified Programming with Dependent Types, by
-  Adam Chlipala.  MIT Press.  2013.  {https://tinyurl.com/zqdnyg2}
+  Adam Chlipala.  MIT Press.  2013.  https://tinyurl.com/zqdnyg2

 [Lipovaca 2011] Learn You a Haskell for Great Good! A Beginner's
   Guide, by Miran Lipovaca, No Starch Press, April 2011.
-  {http://learnyouahaskell.com}
+  http://learnyouahaskell.com

 [O'Sullivan 2008] Bryan O'Sullivan, John Goerzen, and Don Stewart:
   Real world Haskell - code you can believe in. O'Reilly
-  2008. {http://book.realworldhaskell.org}
+  2008. http://book.realworldhaskell.org

 [Pugh 1991] Pugh, William. "The Omega test: a fast and practical
   integer programming algorithm for dependence analysis." Proceedings
   of the 1991 ACM/IEEE conference on Supercomputing. ACM, 1991.
-  {https://dl.acm.org/citation.cfm?id=125848}
+  https://dl.acm.org/citation.cfm?id=125848

 [Wadler 2015] Philip Wadler. "Propositions as types."
   Communications of the ACM 58, no. 12 (2015): 75-84.
-  {https://dl.acm.org/citation.cfm?id=2699407}
+  https://dl.acm.org/citation.cfm?id=2699407

 *)

--- Postscript.v.orig	2025-12-26 15:06:27.901365373 +0100
+++ Postscript.v	2025-12-26 15:06:37.132434717 +0100
@@ -79,7 +79,7 @@
        - For applications of Coq in building verified systems, the
          lectures and course materials for the 2017 DeepSpec Summer
          School are a great resource.
-         {https://deepspec.org/event/dsss17/index.html}
+         https://deepspec.org/event/dsss17/index.html
 *)

 (* 2025-08-20 17:59 *)
--- Induction.v.orig	2025-12-26 15:11:42.721730333 +0100
+++ Induction.v	2025-12-26 15:11:45.877754042 +0100
@@ -53,7 +53,7 @@
        a single archive, a [Makefile] should already exist and you can
        skip this step.)

-         rocq makefile -f _CoqProject *.v -o Makefile
+         rocq makefile -f [_CoqProject] *.v -o Makefile

        You should rerun that command whenever you add or remove
        Rocq files in this directory.
